lightning/src/routing/scoring.rs

```diff
@@ -394,9 +397,48 @@ pub struct ProbabilisticScoringParameters {
 	/// probabilities, the multiplier will have a decreasing effect as the negative `log10` will
 	/// fall below `1`.
 	///
-	/// Default value: 256 msat
+	/// Default value: 192 msat
 	pub liquidity_penalty_amount_multiplier_msat: u64,
 
+	/// A multiplier used in conjunction with the negative `log10` of the channel's success
+	/// probability for the payment, as determined based on the history of our estimates of the
+	/// channel's available liquidity, to determine a penalty.
+	///
+	/// This penalty is similar to [`liquidity_penalty_multiplier_msat`] however instead of using
+	/// only our latest estimate for the current liquidity available in the channel, it estimates
+	/// success probability based on the estimated liquidity available in the channel through
+	/// history. Specifically, every time we update our liquidity bounds on a given channel, we
+	/// track which of several buckets those bounds fall into, exponentially decaying the
+	/// probability of each bucket as new samples are added.
+	///
+	/// Default value: 10,000 msat
+	///
+	/// [`liquidity_penalty_multiplier_msat`]: Self::liquidity_penalty_multiplier_msat
+	pub historical_liquidity_penalty_multiplier_msat: u64,
+
+	/// A multiplier used in conjunction with the payment amount and the negative `log10` of the
+	/// channel's success probability for the payment, as determined based on the history of our
+	/// estimates of the channel's available liquidity, to determine a penalty.
+	///
+	/// The purpose of the amount penalty is to avoid having fees dominate the channel cost for
+	/// large payments. The penalty is computed as the product of this multiplier and the `2^20`ths
+	/// of the payment amount, weighted by the negative `log10` of the success probability.
+	///
+	/// Default value: 64 msat
+	///
+	/// [`liquidity_penalty_multiplier_msat`]: Self::liquidity_penalty_multiplier_msat
+	pub historical_liquidity_penalty_amount_multiplier_msat: u64,
+
+	/// If we aren't learning any new datapoints for a channel, the historical liquidity bounds
+	/// tracking can simply live on with increasingly stale data. Instead, when a channel has not
+	/// seen a liquidity estimate update for this amount of time, the historical datapoints are
+	/// decayed by half.
+	///
+	/// Note that 16 or more half lives guarantees that all historical data will be wiped.
+	///
+	/// Default value: 14 days
```

Comment:
Yes, all I was saying is that *starting* to decay data after two weeks *seems* like a long time to me, as we may not adapt fast enough if something changes. But, as we don't have any ground truth here, it's hard to argue either way. 

Just as a note: What would probably work better than the current model of decaying all at the same rate would be to account for the velocity of behavior changes in the past by, e.g., keeping track of how often our expectation is changing on a per-node basis and decay faster the higher the deviation is. However, this would probably require keeping more data and further introduce the problem how to translate deviation into a corresponding delay. So, we're currently probably better off with the chosen approach.